Source/WebGPU/WebGPU/RenderPassEncoder.mm

```diff
@@ -70,8 +80,23 @@
     [m_renderCommandEncoder endEncoding];
 }
 
+static uint32_t occlusionQuerySetCount()
+{
+    return UINT32_MAX;
+}
+
 void RenderPassEncoder::beginOcclusionQuery(uint32_t queryIndex)
 {
+    if (!prepareTheEncoderState())
+        return;
+
+    if (queryIndex >= occlusionQuerySetCount() || m_occlusionQueryWrites.contains(queryIndex) || m_occlusionQueryActive) {
+        makeInvalid();
+        return;
+    }
```

Comment:
Can we put these in helper functions, with a link to the place in the spec that this is implementing?